### PR TITLE
Add google benchmark to contrib

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Time over time there is a need to write a microbenchmark. Now it requires either to create something ad-hoc with the help of `Stopwatch` or install google benchmark locally, but building process will be anyway complicated.

### CODE REVIEW

File: benchmark/README.md [hunk 1]
```diff
@@ -1 +1,3 @@
-Benchmark is located in a separate repository: https://github.com/ClickHouse/ClickBench
+**Note:** ClickBench is located in a separate repository: https://github.com/ClickHouse/ClickBench
+
+This directory was created for micro-benchmarks (supposedly based on [google benchmark](https://github.com/google/benchmark)).
```
Comment (alexey-milovidov) [inline benchmark/README.md:3]: If we will put microbenchmarks near the code - no need to change this directory.

File: docker/test/fasttest/run.sh [hunk 2]
```diff
@@ -138,6 +138,7 @@ function clone_submodules
             contrib/c-ares
             contrib/morton-nd
             contrib/xxHash
+            contrib/google-benchmark
```
Comment (alexey-milovidov) [inline docker/test/fasttest/run.sh:141]: Why do we need it for fasttest?

**Review by alexey-milovidov — APPROVED** [review]
Everything else LGTM.

**Comment by alexey-milovidov** [comment]
As long as we have ENABLE_TESTS, ENABLE_EXAMPLES, let's add ENABLE_BENCHMARKS.

**Comment by nickitat** [comment]
Stateless tests (tsan) [4/5] - #44038

**Comment by nudles** [comment]
Hi @nickitat, this is a feature that I have been looking for. Thanks!
Actually, I tried to add google benchmark under contrib by myself.
I followed `contrib/abseil-cpp-cmake` to create a `google_benchmark/CMakeLists.txt` file and use `add_subdirectory(google_benchmark_repo_folder)` in it.
However, I got the following issues when running cmake, which seems like that clang cannot find the STL headers.
Does ClickHouse always use the contrib/libcxx and contrib/libcxxabi instead of the STL from the OS? 
Do you know how to tell google_benchmark to use the same STL as ClickHouse? 

```
-- Performing Test HAVE_STD_REGEX -- failed to compile: Change Dir: /data01/fredwang/docker/ClickHouse/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/ninja cmTC_60577 && [1/2] Building CXX object CMakeFiles/cmTC_60577.dir/std_regex.cpp.o
FAILED: CMakeFiles/cmTC_60577.dir/std_regex.cpp.o
/usr/bin/clang++-14 --target=x86_64-linux-gnu --sysroot=/data01/fredwang/docker/ClickHouse/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc   --gcc-toolchain=/data01/fredwang/docker/ClickHouse/cmake/linux/../../contrib/sysroot/linux-x86_64 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/data01/fredwang/docker/ClickHouse=. -falign-functions=32 -mbranches-within-32B-boundaries -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -w  -Wall  -Wextra  -Wshadow  -Wfloat-equal  -Wsuggest-override  -pedantic  -pedantic-errors  -Wshorten-64-to-32  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing  -Wthread-safety  -std=c++11 -MD -MT CMakeFiles/cmTC_60577.dir/std_regex.cpp.o -MF CMakeFiles/cmTC_60577.dir/std_regex.cpp.o.d -o CMakeFiles/cmTC_60577.dir/std_regex.cpp.o -c /data01/fredwang/docker/ClickHouse/contrib/benchmark/cmake/std_regex.cpp
/data01/fredwang/docker/ClickHouse/contrib/benchmark/cmake/std_regex.cpp:1:10: fatal error: 'regex' file not found
#include <regex>
         ^~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.


-- Compiling and running to test HAVE_GNU_POSIX_REGEX
-- cxx
-- Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile: Change Dir: /data01/fredwang/docker/ClickHouse/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/ninja cmTC_31032 && [1/2] Building CXX object CMakeFiles/cmTC_31032.dir/gnu_posix_regex.cpp.o
FAILED: CMakeFiles/cmTC_31032.dir/gnu_posix_regex.cpp.o
/usr/bin/clang++-14 --target=x86_64-linux-gnu --sysroot=/data01/fredwang/docker/ClickHouse/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc   --gcc-toolchain=/data01/fredwang/docker/ClickHouse/cmake/linux/../../contrib/sysroot/linux-x86_64 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/data01/fredwang/docker/ClickHouse=. -falign-functions=32 -mbranches-within-32B-boundaries -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -w  -Wall  -Wextra  -Wshadow  -Wfloat-equal  -Wsuggest-override  -pedantic  -pedantic-errors  -Wshorten-64-to-32  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing  -Wthread-safety  -std=c++11 -MD -MT CMakeFiles/cmTC_31032.dir/gnu_posix_regex.cpp.o -MF CMakeFiles/cmTC_31032.dir/gnu_posix_regex.cpp.o.d -o CMakeFiles/cmTC_31032.dir/gnu_posix_regex.cpp.o -c /data01/fredwang/docker/ClickHouse/contrib/benchmark/cmake/gnu_posix_regex.cpp
/data01/fredwang/docker/ClickHouse/contrib/benchmark/cmake/gnu_posix_regex.cpp:1:10: fatal error: 'gnuregex.h' file not found
#include <gnuregex.h>
         ^~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
```

**Comment by alexey-milovidov** [comment]
> Does ClickHouse always use the contrib/libcxx and contrib/libcxxabi instead of the STL from the OS?

Yes.

The usage of libraries from the OS is not supported at all.

**Comment by nickitat** [comment]
> Do you know how to tell google_benchmark to use the same STL as ClickHouse?

for this we need to provide a custom CMakeLists.txt (`contrib/google-benchmark-cmake/CMakeLists.txt`)

**Pull request merged**